var/spack/repos/builtin/packages/timemory/package.py

```diff
@@ -0,0 +1,133 @@
+    depends_on('mpi', when='+mpi')
+    depends_on('papi', when='+papi')
+    depends_on('cuda', when='+cuda')
+    depends_on('gperftools', when='+gperftools')
+    depends_on('caliper', when='+caliper')
+    depends_on('gperftools', when='+gperftools')
+
+    parallel = True
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        """Set PYTHONPATH to include the site-packages directory for the
+        extension and any other python extensions it depends on."""
+
+        if '+python' in self.spec:
+            # If we set PYTHONHOME, we must also ensure that the corresponding
+            # python is found in the build environment. This to prevent cases
+            # where a system provided python is run against the standard libraries
+            # of a Spack built python. See issue #7128
+            spack_env.set('PYTHONHOME', self.home)
+
+            path = os.path.dirname(self.command.path)
+            if not is_system_path(path):
+                spack_env.prepend_path('PATH', path)
+
+            python_paths = []
+            for d in dependent_spec.traverse(
+                    deptype=('build', 'run', 'test')):
+                if d.package.extends(self.spec):
+                    python_paths.append(join_path(d.prefix,
+                                                  self.site_packages_dir))
+
+            pythonpath = ':'.join(python_paths)
+            spack_env.set('PYTHONPATH', pythonpath)
+
+            # For run time environment set only the path for
+            # dependent_spec and prepend it to PYTHONPATH
+            if dependent_spec.package.extends(self.spec):
+                run_env.prepend_path('PYTHONPATH', join_path(
+                    dependent_spec.prefix, self.site_packages_dir))
+
+    def cmake_args(self):
+        spec = self.spec
+
+        # Use spack install of Caliper instead of internal build
+        args = [
+            "-DTIMEMORY_BUILD_CALIPER=OFF",
+            "-DTIMEMORY_BUILD_TOOLS=ON",
+            "-DTIMEMORY_BUILD_EXTRA_OPTIMIZATIONS=ON",
+            "-DTIMEMORY_BUILD_GTEST=OFF",
+            "-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON",
+        ]
+
+        print("{}".format(self.spec))
+
+        if '+python' in spec:
+            args.append(
+                '-DPYTHON_EXECUTABLE={}'.format(os.path.join(spec['python'].prefix, "bin", "python")))
+            args.append('-DTIMEMORY_BUILD_PYTHON=ON')
+            args.append("-DTIMEMORY_TLS_MODEL='global-dynamic'")
+            python("-m", "pip", "install", "matplotlib", "numpy", "pillow")
+        else:
+            args.append('-DTIMEMORY_BUILD_PYTHON=OFF')
+
+        if '+caliper' in spec:
+            args.append('-DTIMEMORY_USE_CALIPER=ON')
+        else:
+            args.append('-DTIMEMORY_USE_CALIPER=OFF')
+
+        if '+papi' in spec:
+            args.append('-DTIMEMORY_USE_PAPI=ON')
+            args.append('-DPAPI_ROOT_DIR={}'.format(spec['papi'].prefix))
```

Comment:
```suggestion
            args.append('-DPAPI_ROOT_DIR={0}'.format(spec['papi'].prefix))
```
Needed for Python 2.6 support